### ISSUE TITLE
[reviewer one] proof-of-concept validation figure 

### ISSUE THREAD

**Issue by gottacatchenall** [PoisotLab/ms_network_prediction#127]
***Original reviewer comment***

> Considering the above [see #121], validation of proof of concept/and Figure 4 must be updated to reflect the different values/natures of these metrics (e.g., in addition to TSS, a metric from precision-recall space must be shown). The discussion around validation of proof of concept needs to be expanded with: 1) the additional metrics as per above; and 2) backing up of why a TSS around 0.5/50% is actually any good; yes, it should undoubtedly be used instead of accuracy, but what does a roughly around random TSS means? The authors make very strong claims with regards to their proof of concept which need to be backed up. For instance, when measured, proof of concept might yield good performance on the PR-metrics, which could strengthen the argument for it.

**Comment by tpoisot** [comment]
I'm about to infodump, not sorry about this. Hang tight.

**Comment by tpoisot** [comment]
This is the ROC of the new model, with the best classifier highlighted:

![roc-auc](https://user-images.githubusercontent.com/579329/121557559-d64b8f00-c9e2-11eb-8ba8-32a4d44db556.png)

| Measure | Value |
|----|-----|
| AUC | 0.86 |
| Kappa | 0.59 |
| Youden | 0.58 |
| Accuracy | 0.80 |
| Balanced accuracy | 0.79 |

**Comment by tpoisot** [comment]
And these are the loss curves for the testing/training sets:

![loss](https://user-images.githubusercontent.com/579329/121558894-0d6e7000-c9e4-11eb-913f-5ee84a79b21a.png)

Long story short, there is no obvious overfitting and the measures above show that at the right threshold, the model is really quite good.

**Comment by tpoisot** [comment]
I think we might be able to get rid of the validation figure and add a table instead with some explanation of what the validation statistics mean, and then be done with this comment.

**Comment by gottacatchenall** [comment]
^ agreed, so do we also want to replace the accuracy panel with the ROC and loss figure in the proof-of-concept figure 1?

**Comment by tpoisot** [comment]
yup - I'm still working on commenting the code, etc

**Comment by tpoisot** [comment]
I think we might want to show this figure: 

![degree-change](https://user-images.githubusercontent.com/579329/121576286-3992ec80-c9f6-11eb-87ed-024bbe15bde7.png)

It's essentially saying that we do not predict interaction in species that already have a high degree, but rather that we mostly add interactions to parasites with few hosts. 

The consequence is that we are shifting the values of specificity to the left a little bit:

![specificity](https://user-images.githubusercontent.com/579329/121576289-3992ec80-c9f6-11eb-8e00-8e08a7f80bcf.png)

We might want to make a 2-panel figure with these?

**Comment by tpoisot** [comment]
And this is the network with imputed interactions ONLY:

![tsne-imputed](https://user-images.githubusercontent.com/579329/121576490-6e9f3f00-c9f6-11eb-8b65-bd2259f554fb.png)

Just a little way of showing that the predicted interactions are not concentrated at a particular place in the network

**Comment by tpoisot** [comment]
Slight update - imputed interactions in red, original interactions in grey

![tsne-imputed](https://user-images.githubusercontent.com/579329/121577132-364c3080-c9f7-11eb-9857-8aebde2cdcf1.png)

**Comment by tpoisot** [comment]
Fully commented code: https://github.com/PoisotLab/NetworkLearningIllustration/blob/main/main.jl

**Comment by gottacatchenall** [comment]
so how bout this as figure one  

![new_figure1](https://user-images.githubusercontent.com/17133743/121591501-93e87900-ca07-11eb-8cfe-27d7f2bbc0ce.png)

and then a separate figure with 3 panels: degree measured/imputed, specificity density, and the network plot

**Comment by gottacatchenall** [comment]
or potentially the network plot on fig 1 if it fits

**Comment by tpoisot** [comment]
I might make a few suggestions about figure 1 to add some spatial context - will brainstorm on paper and report back

**Comment by gottacatchenall** [comment]
closing this as #130 is more relevant atm
